charts/operator-wandb/templates/ingress.yaml

```diff
@@ -80,3 +81,64 @@ spec:
               number: 8082
   {{- end }}
 {{- end }}
+{{- end }}
+---
+{{- if .Values.gcpPrivatLinkIngress.create }}
+apiVersion: networking.k8s.io/v1
+kind: Ingress
+metadata:
+  name: {{ default .Release.Name .Values.gcpPrivatLinkIngress.nameOverride }}
+  labels:
+    {{- include "wandb.commonLabels" . | nindent 4 }}
+    {{- if .Values.gcpPrivatLinkIngress.labels -}}
+    {{-   toYaml .Values.gcpPrivatLinkIngress.labels | nindent 4 }}
+    {{- end }}
+  annotations:
+    {{- if .Values.gcpPrivatLinkIngress.issuer.create }}
+    "cert-manager.io/issuer": "{{ .Release.Name }}-issuer"
+    "cert-manager.io/acme-challenge-type": "http01"
+    "kubernetes.io/ingress.allow-http" : "false"
+    {{- end }}
+
+    {{- if .Values.gcpPrivatLinkIngress.annotations -}}
+    {{-   toYaml .Values.gcpPrivatLinkIngress.annotations | nindent 4 }}
+    {{- end }}
+spec:
+  ingressClassName: {{ .Values.gcpPrivatLinkIngress.class }}
+  {{- $defaultHost := include "defaultHost" . -}}
+
+  {{- if .Values.gcpPrivatLinkIngress.issuer.create }}
+  tls:
+    - host: [{{ $defaultHost }}]
+      secretName: {{ .Release.Name }}-tls
+  {{- else }}
+  tls: {{ toYaml .Values.gcpPrivatLinkIngress.tls | nindent 4 }}
+  {{- end }}
+
+  rules:
+  {{- range concat (list $defaultHost) .Values.gcpPrivatLinkIngress.additionalHosts }}
+  - host: {{ . }}
+    http:
+      paths:
+      - pathType: Prefix
+        path: /
+        backend:
+          service:
+            {{- if eq $.Values.gcpPrivatLinkIngress.defaultBackend "console" }}
+            name: {{ $.Release.Name }}-console
+            port:
+              number: 8082
+            {{- else }}
+            name: {{ $.Release.Name }}-app
+            port: 
+              number: 8080
+            {{- end }}
+      - pathType: Prefix
+        path: /console
+        backend:
+          service:
+            name: {{ $.Release.Name }}-console
+            port:
+              number: 8082
```

Comment:
Could we move this to a tpl and import that here? This way we only need to modify it in one place